admin-dev/themes/new-theme/js/components/form/submittable-delta-quantity-input.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+import SubmittableInput, {SubmittableInputConfig} from '@components/form/submittable-input';
+import DeltaQuantityInput, {DeltaQuantityConfig} from '@components/form/delta-quantity-input';
+
+type SubmittableConfig = Omit<SubmittableInputConfig, 'wrapperSelector'> & {
+  submittableWrapperSelector: string;
+}
+
+export type SubmittableDeltaConfig = Partial<DeltaQuantityConfig> & SubmittableConfig;
+
+export default class SubmittableDeltaQuantityInput {
+  private deltaQuantityComponent: DeltaQuantityInput;
+
+  private submittableInputComponent: SubmittableInput;
+
+  constructor(deltaConfig: SubmittableDeltaConfig) {
+    const deltaQuantityConfig: Partial<DeltaQuantityConfig> = {};
+
+    if (deltaConfig.containerSelector) {
```

Comment:
It seems a bit complex, there is probably a JS hack to make this simpler But it can be improved later

Reply:
maybe @NeOMakinG aware of a better hack?

Reply:
I think you can just assign properties directly, even by using `Object.assign`

As anyway these properties will be undefined if they are not entering in the condition, so it will cause the same problem later?

There is definitly something to do here